### ISSUE TITLE
Does wasmtime support interpreter mode?

### ISSUE THREAD

**Issue by Zhangyx24** [bytecodealliance/wasmtime#4255]
Does wasmtime support interpreter mode?

**Comment by cfallin** [comment]
Hi @Zhangyx24 -- unfortunately no, we do not currently have an interpreter backend for Wasm execution. This is something we would be open to having, if someone is interested in building it, but currently I'm not aware of anyone who is planning to do so.

I'll go ahead and close this issue since the question is answered but again, we're happy to talk further if someone wants to build this!

**Comment by Zhangyx24** [comment]
Thanks for you answer!
Can I assume that wasmtime supports only JIT and AOT modes?

**Comment by Zhangyx24** [comment]
@cfallin Thanks for your answer!
Can we assume that wasmtime supports only JIT and AOT modes?

**Comment by cfallin** [comment]
(To clarify a bit more: the default "JIT" is an all-at-once compilation at load time, so it does not JIT on-demand as functions call each other, for example. But it is codegen in-process just before running the code, so in that sense it is a JIT.)

**Comment by Zhangyx24** [comment]
Thank you very much! I feel much clearer.

**Comment by hungryzzz** [comment]
> (To clarify a bit more: the default "JIT" is an all-at-once compilation at load time, so it does not JIT on-demand as functions call each other, for example. But it is codegen in-process just before running the code, so in that sense it is a JIT.)

Hi, so even on the JIT mode, the code will be only compiled once, right? And it will not be optimized during execution?

**Comment by bjorn3** [comment]
Indeed. Wasmtime doesn't have multiple optimization tiers, nor speculative optimization. All compilation happens at once when creating the Module from a wasm module.

**Comment by hungryzzz** [comment]
Okay, I got it! Thank you for your answer!
